docs/specs/thrid-party-client-draft.md

```diff
@@ -0,0 +1,38 @@
+## Third-party Clients
+
+The purpose of this document is to shape the third-party client feature. After the details become concrete, this document will be removed and merged to [OIDC](./oidc.md).
+
+## Authorization Flow
+
+- Remove `is_first_party` flag in oauth client, all existing OAuth clients should be first-party apps. Instead, adding `third_party_app` as new `x_application_type`.
+- Third-party client doesn't support anonymous user, biometric login and re-authentication.
+- Allow using `"https://authgear.com/scopes/full-access"` scope for the third-party client, which allows requesting full user's info.
+    - Update checking scope in userinfo endpoint?
+- Update authorization endpoint accepts requests without pkce code challenge
+- Update token endpoint to support using client secret for code exchange
+- (Optional) Support nonce for 3rd party client?
+    - Accept nonce in authz endpoint
+    - Provide nonce in id_token when code exchange
+- Need to add a consent screen?
+    - Since we will only have one full access scope, the message on the consent screen should be general saying Authgear will share the complete user profile to the application.
+    - The consent screen will be shown only if there is no authorization record (first time login / the user revokes it).
+- Open settings page will rely on idp session so that idp session will be created
+    - If the idp session expires, user will be redirect to the login page. They will be redirected back to the settings page after login.
```

Comment:
We fixed this issue long time ago when we were working on first party client. The problem here is, is this the expected behavior?